### ISSUE TITLE
Inconsistent meta tags [SEO]

### ISSUE THREAD

**Issue by mayur-padshala** [wellyshen/react-cool-starter#218]
### Do you want to report a bug or request a feature?

- I want to report a possible bug.

### Expected Behavior

Title and other meta tags should change on first load of the page. This can have adverse effects on my page's SEO behaviour.

### Actual Behavior

- When hitting an error page, the title and meta tag is not changed at first load. When reloading the page, the title changes.
- When going back to a page, the title and meta tag remains the same as in the reloaded error page. This changes when refreshing again.

### Your Environment

- Client OS: Ubuntu Linux (18.04)
- Browser and its version: This is seen on all browsers
- Version of node: 10.15.3

**Comment by wellyshen** [comment]
@mayur-novus This issue doesn't exist in the example pages of this starter kit. I am not sure if it's a bug of the starter kit. Can you check it?

**Comment by aslapkov** [comment]
@wellyshen Hello man! Really, there is a bit problem of the SEO meta tags using `react-helmet` on this starter. After getting the first server rendering response, it shows the wrong meta.
You can check it easily, just open the console **DevTools** -> tab **Network**. Go to the new page, reload this page and choose the needed document, then go to the tab **Response**. Now we can see how looks like the meta tags in fact.

To solve it, we have to use on the server, call `Helmet.renderStatic()` after `renderToString(AppComponent)`. More here https://github.com/nfl/react-helmet#server-usage

And one more thing concerning SEO, we also need to move the lines of code with processing statuses `301` and `404` below, to it's look like that:

```js
if (!__DEV__) {
  // $FlowFixMe: isn't an issue
  const webpackManifest = require('../public/webpack-assets.json');
  assets = bundles
    .map(({ publicPath }) => publicPath)
    .concat(
      Object.keys(webpackManifest)
        .map(key => webpackManifest[key])
        .reverse()
    );
}

// Check if the render result contains a redirect, if so we need to set
// the specific status and redirect header and end the response
if (staticContext.url) {
  res.status(301).setHeader('Location', staticContext.url);
  res.end();

  return;
}

// Check page status
const status = staticContext.status === '404' ? 404 : 200;

// Pass the route and initial state into html template
res
  .status(status)
  .send(renderHtml(head, assets, htmlContent, initialState));
```

It is extremely important to use `staticContext` after the call `renderToString(AppComponent)`.  Otherwise, `staticContext` will not receive the correct status code. This can negatively affect SEO.

**Comment by wellyshen** [comment]
@aslapkov Can you send me a PR for fixing this issue?

**Comment by wellyshen** [comment]
@aslapkov Thank you for your advice. Fix in [v2.8.7](https://github.com/wellyshen/react-cool-starter/releases/tag/v2.8.7).
